### PR TITLE
mocked_host: Document access list behavior

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: include/evmc/mocked_host.hpp [hunk 1]
```diff
@@ -341,6 +352,17 @@ class MockedHost : public Host
     }
 
     /// Access the account's storage value at the given key.
+    ///
+    /// This method is required by EIP-2929 introduced in ::EVMC_BERLIN. In records that the given
+    /// account's storage key has been access and returns the previous access status.
+    /// To mock storage access list, you can pre-init account's storage values with
+    /// ::EVMC_ACCESS_WARM flags:
```
Comment (axic) [inline include/evmc/mocked_host.hpp:359]:
```suggestion
    /// the ::EVMC_ACCESS_WARM flag:
```

File: include/evmc/mocked_host.hpp [hunk 2]
```diff
@@ -322,7 +322,18 @@ class MockedHost : public Host
...
 
     /// Record an account access.
+    ///
+    /// This method is required by EIP-2929 introduced in ::EVMC_BERLIN. It will record the account
+    /// access in MockedHost::recorded_account_accesses and return previous access status.
+    /// This methods returns ::EVMC_ACCESS_WARM for known addresses of precompiles.
+    /// The EIP-2929 specifies that evmc_message::sender and evmc_message::destination are always
+    /// ::EVMC_ACCESS_WARM. Therefore, you should init the MockedHost with
+    /// `access_account(msg.sender)` and `access_account(msg.destination)`.
+    /// The same way you can mock transaction access list for account addresses.
```
Comment (axic) [inline include/evmc/mocked_host.hpp:332]:
```suggestion
    /// The same way you can mock transaction access list (EIP-2930) for account addresses.
```

File: include/evmc/mocked_host.hpp [hunk 3]
```diff
@@ -341,6 +352,17 @@ class MockedHost : public Host
     }
 
     /// Access the account's storage value at the given key.
+    ///
+    /// This method is required by EIP-2929 introduced in ::EVMC_BERLIN. In records that the given
+    /// account's storage key has been access and returns the previous access status.
+    /// To mock storage access list, you can pre-init account's storage values with
```
Comment (axic) [inline include/evmc/mocked_host.hpp:358]:
```suggestion
    /// To mock storage access list (EIP-2930), you can pre-init account's storage values with
```

**Review by axic — APPROVED** [review]

**Comment by chfast** [comment]
Documentation preview: https://32954-66214407-gh.circle-artifacts.com/0/docs/classevmc_1_1MockedHost.html#a51f9acf40acec535549d32c7ab079b9e

File: include/evmc/mocked_host.hpp [hunk 4]
```diff
@@ -341,6 +355,18 @@ class MockedHost : public Host
     }
 
     /// Access the account's storage value at the given key.
+    ///
+    /// This method is required by EIP-2929 introduced in ::EVMC_BERLIN. In records that the given
+    /// account's storage key has been access and returns the previous access status.
+    /// To mock storage access list (EIP-2930), you can pre-init account's storage values with
+    /// the ::EVMC_ACCESS_WARM flags
```
Comment (axic) [inline include/evmc/mocked_host.hpp:362]:
```suggestion
    /// the ::EVMC_ACCESS_WARM flag
```

**Review by axic — APPROVED** [review]

**Pull request merged**